library/item/item_group/src/main/java/org/quiltmc/qsl/item/group/api/QuiltItemGroup.java

```diff
@@ -45,13 +46,15 @@
  */
 public final class QuiltItemGroup extends ItemGroup {
 	private final Supplier<ItemStack> iconSupplier;
+	private Optional<ItemGroupIcon> customIcon;
 	private @Nullable ItemStack icon;
 	private final @Nullable Consumer<List<ItemStack>> stacksForDisplay;
 	private final @Nullable Text displayText;
 
-	private QuiltItemGroup(int index, String id, Supplier<ItemStack> iconSupplier, @Nullable Consumer<List<ItemStack>> stacksForDisplay, @Nullable Text displayText) {
+	private QuiltItemGroup(int index, String id, Supplier<ItemStack> iconSupplier, Optional<ItemGroupIcon> customIcon, @Nullable Consumer<List<ItemStack>> stacksForDisplay, @Nullable Text displayText) {
```

Comment:
I very much don't like that this is limited to just QuiltItemGroups, for two reasons: first of all, it's not very future proof. If we ever want to extend this system, we've got to add stuff to the constructor, so we can't make the constructor public and so people can't extend the functionality of the class. Second of all, it limits anyone wanting to do custom rendering to having to use QuiltItemGroup; especially with #199, I see no reason for this; it also causes pain for anyone doing multiloader stuff.

Reply:
...also, that `Optional<ItemGroupIcon>` should just be a `@Nullable ItemGroupIcon`.